sklearn/impute/_iterative.py

```diff
@@ -344,6 +344,10 @@ def _impute_one_feature(self,
                                      self._min_value[feat_idx],
                                      self._max_value[feat_idx])
 
+        shape_imputed_values = imputed_values.shape
```

Comment:
Please add a [non-regression test](https://en.wikipedia.org/wiki/Non-regression_testing) that would fail at master but pass in this PR.

One possibility is to restart from the example given in the issue: https://github.com/scikit-learn/scikit-learn/issues/19352#issue-801892712

Reply:
@glemaitre some help with the non-regression test please?